cmake/modules/hunter_jobs_number.cmake

```diff
@@ -10,16 +10,36 @@ function(hunter_jobs_number jobs_options_varname toolchain_path)
 
   string(COMPARE EQUAL "${HUNTER_JOBS_NUMBER}" "" use_default_jobs)
   if(use_default_jobs)
+    # NUMBER_OF_LOGICAL_CORES is actually the number of logical cores
+    # per physical core.
+    # Excerpt from the CMake 3.7.1 source code:
+    #     unsigned int GetNumberOfLogicalCPU(); // per physical cpu
+    #     from Source/kwsys/SystemInformation.hxx.in
+    # See https://gitlab.kitware.com/cmake/cmake/issues/16594
     cmake_host_system_information(
-        RESULT jobs_number
+        RESULT l_cores
         QUERY NUMBER_OF_LOGICAL_CORES
     )
-    string(COMPARE EQUAL "${jobs_number}" "0" is_zero)
-    if(is_zero)
-      hunter_status_debug("Number of logical cores is 0 - forcing 1")
-      set(jobs_number "1")
+    cmake_host_system_information(
+        RESULT p_cores
+        QUERY NUMBER_OF_PHYSICAL_CORES
+    )
+    string(COMPARE EQUAL "${l_cores}" "0" l_cores_is_zero)
+    if(l_cores_is_zero)
+      hunter_status_debug("Number of logical cores per physical core is 0 - forcing 1")
+      set(l_cores "1")
     else()
-      hunter_status_debug("Number of logical cores: ${jobs_number}")
+      hunter_status_debug("Number of logical cores per physical core: ${l_cores}")
+    endif()
+    string(COMPARE EQUAL "${p_cores}" "0" p_cores_is_zero)
+    if(p_cores_is_zero)
+      hunter_status_debug("Number of physical cores is 0 - forcing 1")
+      set(p_cores "1")
+    else()
+      hunter_status_debug("Number of physical cores: ${p_cores}")
+    endif()
+    math(EXPR jobs_number "${l_cores} * ${p_cores}")
+    hunter_status_debug("Total number of cores: ${jobs_number}")
     endif()
```

Comment:
Testing info can be found here:
* https://github.com/ruslo/hunter/wiki/dev.hunter.root